### DISCUSSION TITLE
APIs supported in Watch faces and debugging them

### DISCUSSION THREAD

**Discussion by Chriz76** [zepp-health/zeppos-forum#343] [Developer Feedback]
Is there a documentation which APIs are supported in watch faces? I read somewhere that MessageBuilder is not supported. Is that true? Afaik file access is possible (also to data of other apps?). What about BLE?

If e.g. BLE or MessageBuilder are supported, can they be tested in the emulator?

**Comment by MyoungXUE** [comment]
Yes, it is not currently supported. The Watchface has no communication ability except for the file system. The capabilities related to the Watchface are all included in the Watchface API directory.

https://docs.zepp.com/docs/watchface/watchface-quick-start/

**Reply by Chriz76** [reply]
Thank you @MyoungXUE. So I would need a way to modify a file that could be accessed by the watch face. The app-service can't access the filesystem?

It is all about displaying life data from the watch on the watch face.
